src/rpc/rawtransaction_util.h

```diff
@@ -5,16 +5,30 @@
 #ifndef BITCOIN_RPC_RAWTRANSACTION_UTIL_H
 #define BITCOIN_RPC_RAWTRANSACTION_UTIL_H
 
+#include <map>
+
 class CBasicKeyStore;
 class UniValue;
 struct CMutableTransaction;
+class Coin;
+class COutPoint;
 
 namespace interfaces {
 class Chain;
 } // namespace interfaces
 
-/** Sign a transaction with the given keystore and previous transactions */
-UniValue SignTransaction(interfaces::Chain& chain, CMutableTransaction& mtx, const UniValue& prevTxs, CBasicKeyStore *keystore, bool tempKeystore, const UniValue& hashType);
+/**
+ * Sign a transaction with the given keystore and previous transactions
+ *
+ * @param  mtx           The transaction to-be-signed
+ * @param  prevTxs       Array of previous txns outputs that tx depends on but may not yet be in the block chain
+ * @param  keystore      Temporary keystore containing signing keys
+ * @param  coins         Map of unspent outputs - coins in mempool and current chain UTXO set, may be extended by previous txns outputs after call
+ * @param  tempKeystore  Whether to use temporary keystore
+ * @param  hashType      The signature hash type
+ * return signed tx
```

Comment:
`@returns JSON object with details of the signed transaction` or something along that.